nomad/fsm.go

```diff
@@ -518,38 +520,56 @@ func (n *nomadFSM) applyBatchDeregisterJob(buf []byte, index uint64) interface{}
 		panic(fmt.Errorf("failed to decode request: %v", err))
 	}
 
-	for jobNS, options := range req.Jobs {
-		if err := n.handleJobDeregister(index, jobNS.ID, jobNS.Namespace, options.Purge); err != nil {
-			n.logger.Error("deregistering job failed", "job", jobNS, "error", err)
+	// Perform all store updates atomically to ensure a consistent views for store readers.
```

Comment:
views->view

Reply:
Fixed in https://github.com/hashicorp/nomad/commit/5697cbb183342f3b5f8d1d32d70bafc08f1a91be